dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaYarnApplicationsTask.java

```diff
@@ -0,0 +1,177 @@
+  private final int includedLastDays;
+
+  public ClouderaYarnApplicationsTask(int days) {
+    super(String.format("yarn-applications-%d.jsonl", days));
+    this.includedLastDays = days;
+  }
+
+  @Override
+  protected void doRun(
+      TaskRunContext context, @Nonnull ByteSink sink, @Nonnull ClouderaManagerHandle handle)
+      throws Exception {
+    List<ClouderaClusterDTO> clusters = handle.getClusters();
+    Preconditions.checkNotNull(
+        clusters, "Clusters must be initialized before fetching YARN applications.");
+
+    List<ApiYARNApplicationDTO> totalYARNApplications = new ArrayList<>();
+    YarnApplicationsLoader appLoader =
+        new YarnApplicationsLoader(handle.getApiURI().toString(), handle.getHttpClient());
+    for (ClouderaClusterDTO cluster : handle.getClusters()) {
+      String clusterName = cluster.getName();
+      List<String> yarnAppTypes = fetchYARNApplicationTypes(handle, clusterName);
+      for (String yarnAppType : yarnAppTypes) {
+        LOG.info(
+            String.format(
+                "Dump YARN applications with %s type from %s cluster", yarnAppType, clusterName));
+        List<ApiYARNApplicationDTO> yarnApplications = appLoader.load(clusterName, yarnAppType);
+        totalYARNApplications.addAll(yarnApplications);
+        LOG.info(
+            String.format(
+                "Dumped %d YARN applications with %s type from %s cluster",
+                yarnApplications.size(), yarnAppType, clusterName));
+      }
+    }
+
+    String yarnAppsJson = parseObjectToJsonString(totalYARNApplications);
+    try (Writer writer = sink.asCharSink(StandardCharsets.UTF_8).openBufferedStream()) {
+      writer.write(yarnAppsJson);
+    }
+  }
+
+  private String buildISODateTime(int deltaInDays) {
+    ZonedDateTime dateTime =
+        ZonedDateTime.of(LocalDateTime.now().minusDays(deltaInDays), ZoneId.of("UTC"));
+    return dateTime.format(isoDateTimeFormatter);
+  }
+
+  private List<String> fetchYARNApplicationTypes(ClouderaManagerHandle handle, String clusterName) {
+    List<String> yarnApplicationTypes = new ArrayList<>(predefinedAppTypes);
+    String yarnAppTypesUrl =
+        handle.getApiURI().toString() + "clusters/" + clusterName + "/serviceTypes";
+    CloseableHttpClient httpClient = handle.getHttpClient();
+    JsonNode yarnAppTypesJson;
+    ApiYARNApplicationTypeDTO yarnAppTypesFromClouderaAPI;
+    try (CloseableHttpResponse chart = httpClient.execute(new HttpGet(yarnAppTypesUrl))) {
+      yarnAppTypesJson = readJsonTree(chart.getEntity().getContent());
+      yarnAppTypesFromClouderaAPI =
+          parseJsonStringToObject(yarnAppTypesJson.toString(), ApiYARNApplicationTypeDTO.class);
+      yarnApplicationTypes.addAll(yarnAppTypesFromClouderaAPI.getItems());
+    } catch (IOException ex) {
+      throw new RuntimeException(ex.getMessage(), ex);
+    }
+    return yarnApplicationTypes;
+  }
+
+  private class YarnApplicationsLoader {
```

Comment:
We don't need this class. As I see the logic of `ClouderaYarnApplicationsTask`:
For each cluster:
1. fetch all application types
2. fetch all jobs per application type with pagination. 

So, it's more like 2 private method of this task but not class.

Reply:
I added this class to simplify amount of parameters which should be passed to the private methods. As we need to pass pagination data (offset, limit), common parameters (URL host, httpClient), clusterName, application type.

Reply:
This class has been moved into the abstract class.